apps/keyboard/js/views/candidate_panel_view.js

```diff
@@ -0,0 +1,249 @@
+'use strict';
+
+/* global BaseView */
+
+(function(exports) {
+
+/**
+ * CandidateView handles the rendering of each candidate, i.e.
+ * the word suggestion.
+ */
+function CandidateView() {
+  BaseView.apply(this, arguments);
+}
+
+CandidateView.prototype = Object.create(BaseView.prototype);
+
+CandidateView.prototype.render = function render() {
+  var span = document.createElement('span');
+  span.textContent = this.target.text;
+  span.style.flex = this.options.unit;
+  // ui/integration test needs this
+  span.dataset.data = this.target.data;
+
+  this.element = span;
+
+  if (this.viewManager) {
+    this.viewManager.registerView(this.target, this);
+  }
+};
+
+/**
+ * CandidateToggleButtonView represents the toggle button to show/hide
+ * the full candidate panel.
+ */
+function CandidateToggleButtonView() {
+  BaseView.apply(this, arguments);
+}
+
+CandidateToggleButtonView.prototype = Object.create(BaseView.prototype);
+
+CandidateToggleButtonView.prototype.render = function() {
+  var toggleButton = document.createElement('div');
+  toggleButton.classList.add('keyboard-candidate-panel-toggle-button');
+  toggleButton.classList.add('hide');
+
+  this.element = toggleButton;
+
+  if (this.viewManager) {
+    this.viewManager.registerView(this.target, this);
+  }
+};
+
+CandidateToggleButtonView.prototype.show = function() {
+  this.element.classList.remove('hide');
+};
+
+CandidateToggleButtonView.prototype.hide = function() {
+  this.element.classList.add('hide');
+};
+
+/**
+ * CandidatePanelView handles the rendering of candidate panel, used for showing
+ * word suggestions.
+ */
+function CandidatePanelView(target, options, viewManager) {
+  this.element = null;
+  this.className = options.className;
+  this.viewManager = viewManager;
```

Comment:
Let's let all views inherit `BaseView` instead so you don't have to repeat here.